cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -84,6 +86,9 @@ public KafkaBrokerConfigurationBuilder withCruiseControl(String clusterName, Cru
             writer.println("cruise.control.metrics.reporter.ssl.truststore.type=PKCS12");
             writer.println("cruise.control.metrics.reporter.ssl.truststore.location=/tmp/kafka/cluster.truststore.p12");
             writer.println("cruise.control.metrics.reporter.ssl.truststore.password=${CERTS_STORE_PASSWORD}");
+            writer.println("cruise.control.metrics.topic.auto.create=true");
+            writer.println("cruise.control.metrics.topic.num.partitions=" + defaultNumPartitions);
+            writer.println("cruise.control.metrics.topic.replication.factor=" + defaultReplicationFactor);
```

Comment:
I am a little bit confused here. We are enabling the metrics reporter topic auto-creation but we actually know that it's not going to work when Kafka broker auto topic creation is disabled. 
Regarding the num of partitions and replication factor, I see that these parameters are coming from the `KafkaCluster` model where the defaults are "1". It sounds weird because they should be some defaults for specific metrics reporter topic creation not related to Kafka itself (which has these values by default to 1 anyway when the auto topic creation is enabled).
We are also enabling this stuff by default, is really what we want? It means that user cannot disable metrics reporter auto topic creation, or?

Reply:
> I am a little bit confused here. We are enabling the metrics reporter topic auto-creation but we actually know that it's not going to work when Kafka broker auto topic creation is disabled.

As discussed earlier today, the metrics reporter topic auto-creation will work in all situations except when being started with a new Kafka cluster, so we enable by default and forbid the user from setting it. This way, the topic creation/configuration will work correctly for all other situations

> Regarding the num of partitions and replication factor, I see that these parameters are coming from the KafkaCluster model where the defaults are "1". It sounds weird because they should be some defaults for specific metrics reporter topic creation not related to Kafka itself (which has these values by default to 1 anyway when the auto topic creation is enabled).

Ah yes, those values should be coming from CC metric reporter config first and the Kafka config  second if the CC metric reporter config are not set for these fields. I'll update that in the next commit, nice catch!

> We are also enabling this stuff by default, is really what we want? It means that user cannot disable metrics reporter auto topic creation, or?

Yes definitely, we only gain by having this setting locked in as enabled, plus we will this enabled anyway for the metric reporter patch will commit upstream.